### PR TITLE
Disable mmap for StorageFile in clickhouse-server

### LINKED ISSUE #47159: SIGBUS in `segmentatorThreadFunction`/`fileSegmentationEngine*`
https://s3.amazonaws.com/clickhouse-test-reports/46681/ad4a44df52b6b04ac5977d12aa35b099a792133c/upgrade_check__asan_.html 

```
2023-03-02 07:45:26 Thread 1305 "Segmentator" received signal SIGBUS, Bus error.
2023-03-02 07:45:26 [Switching to Thread 0x7f933171b700 (LWP 139986)]
2023-03-02 07:45:26 detail::find_first_symbols_sse2<true, (detail::ReturnMode)0, (char)13, (char)10> (begin=<optimized out>, end=0x7f9a24a18095 "8)\003\002") at ../base/base/find_symbols.h:85
2023-03-02 07:45:26 #0  detail::find_first_symbols_sse2<true, (detail::ReturnMode)0, (char)13, (char)10> (begin=<optimized out>, end=0x7f9a24a18095 "8)\003\002") at ../base/base/find_symbols.h:85
2023-03-02 07:45:26         bit_mask = <optimized out>
2023-03-02 07:45:26         pos = 0x7f9a24a18000 "\002\anested1\anested20Array(Tuple(Array(UInt64), Map(String, UInt64)))TTuple(Tuple(Array(Array(UInt64)), Map(UInt64, Array(Tuple(UInt64, String)))), UInt8)\003\002"
2023-03-02 07:45:26         bit_mask = <optimized out>
2023-03-02 07:45:26         bytes = <optimized out>
2023-03-02 07:45:26         eq = <optimized out>
2023-03-02 07:45:26 #1  detail::find_first_symbols_dispatch<true, (detail::ReturnMode)0, (char)13, (char)10> (begin=<optimized out>, end=0x7f9a24a18095 "8)\003\002") at ../base/base/find_symbols.h:194
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #2  find_first_symbols<(char)13, (char)10> (begin=<optimized out>, end=0x7f9a24a18095 "8)\003\002") at ../base/base/find_symbols.h:211
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #3  DB::fileSegmentationEngineTabSeparatedImpl (in=..., memory=..., is_raw=<optimized out>, min_bytes=19425859, min_rows=3, max_rows=19348) at ../src/Processors/Formats/Impl/TabSeparatedRowInputFormat.cpp:377
2023-03-02 07:45:26         need_more_data = true
2023-03-02 07:45:26         pos = <optimized out>
2023-03-02 07:45:26         number_of_rows = <optimized out>
2023-03-02 07:45:26 #4  0x0000000014865978 in std::__1::__function::__policy_func<std::__1::pair<bool, unsigned long> (DB::ReadBuffer&, DB::Memory<Allocator<false, false> >&, unsigned long, unsigned long)>::operator()[abi:v15000](DB::ReadBuffer&, DB::Memory<Allocator<false, false> >&, unsigned long&&, unsigned long&&) const (this=0x7f97b21eacb0, __args=<optimized out>, __args=<optimized out>, __args=<optimized out>, __args=<optimized out>) at ../contrib/llvm-project/libcxx/include/__functional/function.h:848
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #5  std::__1::function<std::__1::pair<bool, unsigned long> (DB::ReadBuffer&, DB::Memory<Allocator<false, false> >&, unsigned long, unsigned long)>::operator()(DB::ReadBuffer&, DB::Memory<Allocator<false, false> >&, unsigned long, unsigned long) const (this=0x7f97b21eacb0, __arg=19348, __arg=19348, __arg=19348, __arg=19348) at ../contrib/llvm-project/libcxx/include/__functional/function.h:1187
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #6  DB::ParallelParsingInputFormat::segmentatorThreadFunction (this=0x7f97b21eab18, thread_group=...) at ../src/Processors/Formats/Impl/ParallelParsingInputFormat.cpp:41
2023-03-02 07:45:26         segmentator_unit_number = <optimized out>
2023-03-02 07:45:26         unit = <optimized out>
2023-03-02 07:45:26         have_more_data = <optimized out>
2023-03-02 07:45:26         currently_read_rows = <optimized out>
2023-03-02 07:45:26         scope_exit16 = {static is_nullable = <optimized out>, function = {thread_group = @0x7f9331712748}}
2023-03-02 07:45:26 #7  0x0000000014869ad6 in std::__1::__invoke[abi:v15000]<void (DB::ParallelParsingInputFormat::*&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*&, std::__1::shared_ptr<DB::ThreadGroupStatus>&, void> (__f=<optimized out>, __a0=<optimized out>, __args=...) at ../contrib/llvm-project/libcxx/include/__functional/invoke.h:359
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #8  std::__1::__apply_tuple_impl[abi:v15000]<void (DB::ParallelParsingInputFormat::*&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), std::__1::tuple<DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >&, 0ul, 1ul>(void (DB::ParallelParsingInputFormat::*&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), std::__1::tuple<DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >&, std::__1::__tuple_indices<0ul, 1ul>) (__f=<optimized out>, __t=...) at ../contrib/llvm-project/libcxx/include/tuple:1789
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #9  std::__1::apply[abi:v15000]<void (DB::ParallelParsingInputFormat::*&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), std::__1::tuple<DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >&>(void (DB::ParallelParsingInputFormat::*&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), std::__1::tuple<DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >&) (__f=<optimized out>, __t=...) at ../contrib/llvm-project/libcxx/include/tuple:1798
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #10 ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<void (DB::ParallelParsingInputFormat::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >(void (DB::ParallelParsingInputFormat::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&)::{lambda()#1}::operator()() (this=0x7f9983847c00) at ../src/Common/ThreadPool.h:210
2023-03-02 07:45:26         thread_status = <incomplete type>
2023-03-02 07:45:26         scope_exit198 = {static is_nullable = false, function = {<No data fields>}}
2023-03-02 07:45:26         function = <optimized out>
2023-03-02 07:45:26         arguments = {__base_ = {<std::__1::__tuple_leaf<0ul, DB::ParallelParsingInputFormat*, false>> = {__value_ = <optimized out>}, <std::__1::__tuple_leaf<1ul, std::__1::shared_ptr<DB::ThreadGroupStatus>, false>> = {__value_ = {__ptr_ = 0x7f996558a618, __cntrl_ = 0x7f996558a600}}, <No data fields>}}
2023-03-02 07:45:26 #11 std::__1::__invoke[abi:v15000]<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<void (DB::ParallelParsingInputFormat::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >(void (DB::ParallelParsingInputFormat::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&)::{lambda()#1}&> (__f=...) at ../contrib/llvm-project/libcxx/include/__functional/invoke.h:394
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #12 std::__1::__invoke_void_return_wrapper<void, true>::__call<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<void (DB::ParallelParsingInputFormat::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >(void (DB::ParallelParsingInputFormat::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&)::{lambda()#1}&>(ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<void (DB::ParallelParsingInputFormat::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >(void (DB::ParallelParsingInputFormat::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&)::{lambda()#1}&) (__args=...) at ../contrib/llvm-project/libcxx/include/__functional/invoke.h:479
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #13 std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<void (DB::ParallelParsingInputFormat::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >(void (DB::ParallelParsingInputFormat::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&)::{lambda()#1}, void ()>::operator()[abi:v15000]() (this=0x7f9983847c00) at ../contrib/llvm-project/libcxx/include/__functional/function.h:235
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #14 std::__1::__function::__policy_invoker<void ()>::__call_impl<std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<void (DB::ParallelParsingInputFormat::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >(void (DB::ParallelParsingInputFormat::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&)::{lambda()#1}, void ()> >(std::__1::__function::__policy_storage const*) (__buf=<optimized out>) at ../contrib/llvm-project/libcxx/include/__functional/function.h:716
2023-03-02 07:45:26         __f = 0x7f9983847c00
2023-03-02 07:45:26 #15 0x000000000e194dea in std::__1::__function::__policy_func<void ()>::operator()[abi:v15000]() const (this=0x7f9331712a60) at ../contrib/llvm-project/libcxx/include/__functional/function.h:848
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #16 std::__1::function<void ()>::operator()() const (this=0x7f9331712a60) at ../contrib/llvm-project/libcxx/include/__functional/function.h:1187
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #17 ThreadPoolImpl<std::__1::thread>::worker (this=0x7f9a2399d2c0, thread_it=...) at ../src/Common/ThreadPool.cpp:315
2023-03-02 07:45:26         metric_active_threads = {what = <optimized out>, amount = 1}
2023-03-02 07:45:26         thread_trace_context = {root_span = {trace_id = {t = {items = {0, 0}}}, span_id = 0, parent_span_id = 0, operation_name = {static __endian_factor = 1, __r_ = {<std::__1::__compressed_pair_elem<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::__rep, 0, false>> = {__value_ = {{__l = {__data_ = 0x0, __size_ = 0, __cap_ = 0, __is_long_ = 0}, __s = {__data_ = '\000' <repeats 22 times>, __padding_ = 0x7f9331712b47 "", __size_ = 0 '\000', __is_long_ = 0 '\000'}, __r = {__words = {0, 0, 0}}}}}, <std::__1::__compressed_pair_elem<std::__1::allocator<char>, 1, true>> = {<std::__1::allocator<char>> = {<std::__1::__non_trivial_if<true, std::__1::allocator<char> >> = {<No data fields>}, <No data fields>}, <No data fields>}, <No data fields>}, static npos = 18446744073709551615}, start_time_us = 0, finish_time_us = 0, attributes = {<std::__1::vector<DB::Field, AllocatorWithMemoryTracking<DB::Field> >> = {__begin_ = 0x0, __end_ = 0x0, __end_cap_ = {<std::__1::__compressed_pair_elem<DB::Field*, 0, false>> = {__value_ = 0x0}, <std::__1::__compressed_pair_elem<AllocatorWithMemoryTracking<DB::Field>, 1, true>> = {<AllocatorWithMemoryTracking<DB::Field>> = {<No data fields>}, <No data fields>}, <No data fields>}}, <No data fields>}}, is_context_owner = true}
2023-03-02 07:45:26         job = {<std::__1::__function::__maybe_derive_from_unary_function<void ()>> = {<No data fields>}, <std::__1::__function::__maybe_derive_from_binary_function<void ()>> = {<No data fields>}, __f_ = {__buf_ = {__small = "\000|\204\203\231\177\000\000\000\336\226\r\231\177\000", __large = 0x7f9983847c00}, __invoker_ = {__call_ = 0x14869a60 <std::__1::__function::__policy_invoker<void ()>::__call_impl<std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<void (DB::ParallelParsingInputFormat::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >(void (DB::ParallelParsingInputFormat::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&)::{lambda()#1}, void ()> >(std::__1::__function::__policy_storage const*)>}, __policy_ = 0x4dcebe8 <std::__1::__function::__policy::__choose_policy[abi:v15000]<std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<void (DB::ParallelParsingInputFormat::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*, std::__1::shared_ptr<DB::ThreadGroupStatus> >(void (DB::ParallelParsingInputFormat::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>), DB::ParallelParsingInputFormat*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&)::{lambda()#1}, void ()> >(std::__1::integral_constant<bool, false>)::__policy_>}}
2023-03-02 07:45:26         parent_thead_trace_context = {<DB::OpenTelemetry::TracingContext> = {trace_id = {t = {items = {0, 0}}}, span_id = 0, tracestate = {static __endian_factor = 1, __r_ = {<std::__1::__compressed_pair_elem<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::__rep, 0, false>> = {__value_ = {{__l = {__data_ = 0x0, __size_ = 0, __cap_ = 0, __is_long_ = 0}, __s = {__data_ = '\000' <repeats 22 times>, __padding_ = 0x7f9331712aef "", __size_ = 0 '\000', __is_long_ = 0 '\000'}, __r = {__words = {0, 0, 0}}}}}, <std::__1::__compressed_pair_elem<std::__1::allocator<char>, 1, true>> = {<std::__1::allocator<char>> = {<std::__1::__non_trivial_if<true, std::__1::allocator<char> >> = {<No data fields>}, <No data fields>}, <No data fields>}, <No data fields>}, static npos = 18446744073709551615}, trace_flags = 0 '\000'}, span_log = {__ptr_ = 0x0, __cntrl_ = 0x0}}
2023-03-02 07:45:26         need_shutdown = true
2023-03-02 07:45:26         metric_all_threads = {what = <optimized out>, amount = 1}
2023-03-02 07:45:26 #18 0x000000000e19a4a1 in ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}::operator()() const (this=0x7f98661b71c8) at ../src/Common/ThreadPool.cpp:145
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #19 std::__1::__invoke[abi:v15000]<ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}> (__f=...) at ../contrib/llvm-project/libcxx/include/__functional/invoke.h:394
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #20 std::__1::__thread_execute[abi:v15000]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>&, std::__1::__tuple_indices<>) (__t=...) at ../contrib/llvm-project/libcxx/include/thread:284
2023-03-02 07:45:26 No locals.
2023-03-02 07:45:26 #21 std::__1::__thread_proxy[abi:v15000]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}> >(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>) (__vp=0x7f98661b71c0) at ../contrib/llvm-project/libcxx/include/thread:295
2023-03-02 07:45:26         __p = {__ptr_ = {<std::__1::__compressed_pair_elem<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, (lambda at ../src/Common/ThreadPool.cpp:145:42)> *, 0, false>> = {__value_ = 0x7f98661b71c0}, <std::__1::__compressed_pair_elem<std::__1::default_delete<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, (lambda at ../src/Common/ThreadPool.cpp:145:42)> >, 1, true>> = {<std::__1::default_delete<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, (lambda at ../src/Common/ThreadPool.cpp:145:42)> >> = {<No data fields>}, <No data fields>}, <No data fields>}}
2023-03-02 07:45:26 #22 0x00007f9a249ea609 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
2023-03-02 07:45:26 No symbol table info available.
2023-03-02 07:45:26 #23 0x00007f9a2490f133 in clone () from /lib/x86_64-linux-gnu/libc.so.6
2023-03-02 07:45:26 No symbol table info available.
```

More reports: [link](https://play.clickhouse.com/play?user=play#d2l0aCAKJyUlJyBhcyBuYW1lX3BhdHRlcm4sCiclZmlsZVNlZ21lbnRhdGlvbkVuZ2luZSUnIGFzIGNvbnRleHRfcGF0dGVybiwKJzIwMjItMDktMDEnIGFzIHN0YXJ0X2RhdGUsCig0NTQ2MSkgYXMgbm9pc3lfcHJzLAooJ1N0YXRlbGVzcyB0ZXN0cyAoYXNhbiknLCAnU3RhdGVsZXNzIHRlc3RzIChhZGRyZXNzKScsICdTdGF0ZWxlc3MgdGVzdHMgKGFkZHJlc3MsIGFjdGlvbnMpJykgYXMgYmFja3BvcnRfYW5kX3JlbGVhc2Vfc3BlY2lmaWNfY2hlY2tzCnNlbGVjdCAKdG9TdGFydE9mRGF5KGNoZWNrX3N0YXJ0X3RpbWUpIGFzIGQsCmNvdW50KCksICBncm91cFVuaXFBcnJheShwdWxsX3JlcXVlc3RfbnVtYmVyKSwgYW55KHJlcG9ydF91cmwpCmZyb20gY2hlY2tzIHdoZXJlIHN0YXJ0X2RhdGUgPD0gY2hlY2tfc3RhcnRfdGltZSBhbmQgcHVsbF9yZXF1ZXN0X251bWJlciBub3QgaW4gCihzZWxlY3QgcHVsbF9yZXF1ZXN0X251bWJlciBhcyBwcm4gZnJvbSBjaGVja3Mgd2hlcmUgcHJuIT0wIGFuZCBzdGFydF9kYXRlIDw9IGNoZWNrX3N0YXJ0X3RpbWUgYW5kIGNoZWNrX25hbWUgaW4gYmFja3BvcnRfYW5kX3JlbGVhc2Vfc3BlY2lmaWNfY2hlY2tzKSAKYW5kIHRlc3RfbmFtZSBsaWtlIG5hbWVfcGF0dGVybiBhbmQgdGVzdF9jb250ZXh0X3JhdyBpbGlrZSBjb250ZXh0X3BhdHRlcm4KYW5kIHB1bGxfcmVxdWVzdF9udW1iZXIgbm90IGluIG5vaXN5X3BycwphbmQgdGVzdF9zdGF0dXMgaW4gKCdGQUlMJywgJ0ZMQUtZJywgJ2ZhaWx1cmUnKSBncm91cCBieSBkIG9yZGVyIGJ5IGQgZGVzYw==)

cc: @nikitamikhaylov

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

This closes #47159.

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit 8a23d8b38925a6b29bedcb60f142871cb3e09a1f with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49717/8a23d8b38925a6b29bedcb60f142871cb3e09a1f/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by vmarora — APPROVED** [review]

**Review by Avogar — APPROVED** [review]

**Comment by Avogar** [comment]
Need to update the test. I will do it

File: src/Storages/StorageFile.cpp [hunk 1]
```diff
@@ -201,7 +201,19 @@ std::unique_ptr<ReadBuffer> selectReadBuffer(
...
+      * - IO error from the device will result in SIGBUS;
+      * - recovery from this signal is not feasible even with the usage of siglongjmp,
+      *   as it might require stack unwinding from arbitrary place;
+      * - arbitrary slowdown due to page fault in arbitrary place in the code is difficult to debug.
+      *
+      * But we keep this mode for clickhouse-local as it is not so bad for a command line tool.
+      */
+
+    if (S_ISREG(file_stat.st_mode)
+        && context->getApplicationType() != Context::ApplicationType::SERVER
```
Comment (azat) [inline src/Storages/StorageFile.cpp:215]: Maybe it worth to throw in error instead of simply ignore it?

Comment (Avogar) [reply src/Storages/StorageFile.cpp:215]: But we want `storage_file_read_method=mmap` by default in `clickhouse-local`. So, the default value is still `mmap`, but in server we fallback to `pread`

Comment (Avogar) [reply src/Storages/StorageFile.cpp:215]: It will be better to have 2 defaults, 1 for server and 1 for local, but I am not sure if we can do it gracefully

Comment (azat) [reply src/Storages/StorageFile.cpp:215]: Oh, I see, make sense.

Though simply ignoring it also looks icky, maybe at least a `LOG_WARNING`, or having different defaults for `clickhouse-local` (I thought about enabling `allow_introspection_functions` for `clickhouse-local` by default).

Another option, for this particular setting is to add `auto`, that will depends on the application type.

Thoughts?

Comment (Avogar) [reply src/Storages/StorageFile.cpp:215]: I like the idea of separate defaults for server and local

Comment (alexey-milovidov) [reply src/Storages/StorageFile.cpp:215]: I also like it.

Comment (azat) [reply src/Storages/StorageFile.cpp:215]: - #51363

**Pull request merged**